doctr/transforms/modules/pytorch.py

```diff
@@ -27,26 +28,78 @@ def __init__(
         self.preserve_aspect_ratio = preserve_aspect_ratio
         self.symmetric_pad = symmetric_pad
 
-    def forward(self, img: torch.Tensor) -> torch.Tensor:
-        target_ratio = self.size[0] / self.size[1]
-        actual_ratio = img.shape[-2] / img.shape[-1]
-        if not self.preserve_aspect_ratio or (target_ratio == actual_ratio):
+    def forward(
+        self,
+        img: torch.Tensor,
+        target: Optional[np.ndarray] = None,
+    ) -> Union[torch.Tensor, Tuple[torch.Tensor, np.ndarray]]:
```

Comment:
Thinking about it, I think we should have:
- one function to resize the image
- one function to resize the target
- use them in the module implementation
And I'm not sure what would be the best, but probably having `Resize` (image only), and `SampleResize` (image + target), would be relevant. What do you think?

Reply:
Yes we can split the function with 1 for the targets and 1 for the image, but I am not sure it is relevant to have a` Resize` and a `SampleResize`, because if you want to keep the aspect ratio of the image while resizing, you will almost always pad (except if you give only 1 target size), and this will modify the targets. The ` Resize`  function won't be able to preserve the aspect ratio of the images, and this would need to differentiate the function we are using in the training scripts and in the preprocessor regarding the option selected by the user (`preserve_aspect_ratio` or not). This can even be dangerous if someone tries to modify the aspect ratio in ` Resize`  without changing the targets I think, do yo agree ?

Reply:
And for a differentiation of the target and image function, I think it will complexify the code because we use many attributes of the `Resize` class to resize both the image and the target (`self.symmetric_pad`, `self.preserve_aspect_ratio`, `self.output_size`) which would be passed as arguments and we have also many shared computations for the image and the target that would need to be done twice (or added in the signature of the functions but I think it will make the code less understandable): `offset`,` raw_shape,` and `img.shape` is also used in the target computation, what do you think ?

Reply:
About splitting the two transforms, the reason behind this is that the target, depending on the task, will not necessarily be modified. So either we can add a lot of cases (for each target type) in the Resize transform, or we could have a Resize that only transforms the image, and `SampleResize` that inherits from it and transforms the target.

I agree this would be dangerous for something to add this without changing the target, but transforms will only be played with by people willing to train models. So I would argue it's safe to assume they either use our default training script or have a good understanding of what is going on under the hood :man_shrugging: 

However, I fully agree about the duplication of symmetric pad / computation duplication for some aspects. I don't have an ideal suggestion for this

Reply:
For the `SampleResize` transform, you want it to inherit from `Resize`, but since they won't have the same signature so I don't really see why we do that ? @fg-mindee

Reply:
We can always refactor this later on anyway! Feel free to implement it the way you prefer (the optional target passing you suggested is probably the best one)